src/utils/exithandler/exithandler.cpp

```diff
@@ -0,0 +1,41 @@
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
+#include "exithandler.h"
+
+#include <signal.h>
+#include <src/utils/text/text.h>
+
+QCoreApplication* ExitHandler::_app = NULL;
+int ExitHandler::_counter = 0;
+
+ExitHandler::ExitHandler() {
+
+}
+
+void ExitHandler::setApplication(QApplication *app) {
+    ExitHandler::_app = app;
+}
+
+void ExitHandler::setup() {
+    struct sigaction sigIntHandler;
+    sigIntHandler.sa_handler = ExitHandler::run;
+    sigemptyset(&sigIntHandler.sa_mask);
+    sigIntHandler.sa_flags = 0;
+    sigaction(SIGINT, &sigIntHandler, NULL);
+}
+
+void ExitHandler::run(int s) {
+    ExitHandler::_counter++;
+    switch(ExitHandler::_counter) {
```

Comment:
With this, every time that the application is closed, the run() method is called, so we can make an more efficient management of the application close events